src/routes/news/components/NewsList.js

```diff
@@ -0,0 +1,251 @@
+  isFetchingMore: boolean,
+  fetchMoreNews: () => void,
+  navigateToNews: (navigationOptions: NavigateToNewsParamsType) => void,
+  createNavigateToNews: (NavigateToNewsParamsType) => void
+|}
+
+/**
+ * Displays a list of news or a single news item, matching the route <id>)
+ */
+class NewsList extends React.PureComponent<PropsType> {
+  navigateToNews = (cityCode: string, language: string, path: string) => () => {
+    const { selectedNewsType } = this.props
+    this.props.navigateToNews({
+      cityCode,
+      language,
+      path,
+      type: selectedNewsType
+    })
+  };
+
+  openTunewsLink = async () => {
+    const supported = await Linking.canOpenURL(tunewsWebsiteUrl)
+    if (supported) {
+      await Linking.openURL(tunewsWebsiteUrl)
+    }
+  };
+
+  renderNoItemsComponent = () => {
+    const { t, theme } = this.props
+    return <NoNews theme={theme}>{t('currentlyNoNews')}</NoNews>
+  };
+
+  rendersNewsListItem = (cityCode: string, language: string) => ({
+    item: newsItem
+  }) => {
+    const { theme, selectedNewsType } = this.props
+    const isTunews = selectedNewsType === TUNEWS
+    return (
+      <NewsListItem
+        key={newsItem.id}
+        newsItem={newsItem}
+        language={language}
+        theme={theme}
+        isTunews={isTunews}
+        navigateToNews={this.navigateToNews(
+          cityCode,
+          language,
+          `${newsItem.id}`
+        )}
+      />
+    )
+  };
+
+  render () {
+    const {
+      newsList,
+      path,
+      cityCode,
+      language,
+      theme,
+      t,
+      fetchMoreNews,
+      isFetchingMore,
+      selectedNewsType
+    } = this.props
+
+    const isTunews = selectedNewsType === TUNEWS
+
+    if (path) {
+      const selectedNewsItem: LocalNewsModel | TunewsModel = newsList.find(
+        _newsItem => `${_newsItem.id}` === path
+      )
+
+      if (selectedNewsItem) {
+        const { eNewsNo, date, content, message } = selectedNewsItem
+
+        return (
+          <View style={{ flex: 1 }}>
+            <ScrollView
+              showsVerticalScrollIndicator={false}
+              contentContainerStyle={{
+                flexGrow: 1,
+                marginBottom: 10,
+                paddingHorizontal: '5%'
+              }}>
+              {isTunews && (
+                <HeaderImageWrapper>
+                  <HeaderImage source={headerImage} />
+                </HeaderImageWrapper>
+              )}
+              <Container>
+                <NewsDetailsTitle theme={theme}>
+                  {selectedNewsItem.title}
+                </NewsDetailsTitle>
+                <NewsDetailsContent theme={theme} language={language}>
+                  {content || message}
+                </NewsDetailsContent>
+              </Container>
+              <Row theme={theme} language={language}>
+                {isTunews ? (
+                  <ExtraInfo theme={theme}>
+                    {`${t('eNewsNo')}: ${eNewsNo}`}
```

Comment:
We are still discussing this but maybe we won't need the `eNewsNo` string description. I'll get back to you.